targets/ublksrv_tgt.cpp

```diff
@@ -3,159 +3,16 @@
 #include "config.h"
 #include "ublksrv_tgt.h"
 
-#define ERROR_EVTFD_DEVID   0xfffffffffffffffe
-
-static void ublksrv_ctrl_data_init(struct ublksrv_ctrl_dev *cdev,
-		struct ublksrv_ctrl_data *data, bool recover)
-{
-	ublksrv_tgt_jbuf_init(cdev, &data->jbuf, recover);
-	sem_init(&data->queue_sem, 0, 0);
-	data->recover = recover;
-	ublksrv_ctrl_set_priv_data(cdev, data);
-}
-
-static void ublksrv_ctrl_data_exit(struct ublksrv_ctrl_dev *cdev,
-		struct ublksrv_ctrl_data *data)
-{
-	ublksrv_ctrl_set_priv_data(cdev, NULL);
-	ublksrv_tgt_jbuf_exit(&data->jbuf);
-}
-
-struct ublksrv_tgt_jbuf *ublksrv_tgt_get_jbuf(const struct ublksrv_ctrl_dev *cdev)
-{
-	struct ublksrv_ctrl_data *data = ublksrv_get_ctrl_data(cdev);
-
-	return &data->jbuf;
-}
-
-int ublk_json_write_dev_info(const struct ublksrv_ctrl_dev *cdev)
-{
-	struct ublksrv_tgt_jbuf *j = ublksrv_tgt_get_jbuf(cdev);
-	int ret = 0;
-
-	if (!j)
-		return -EINVAL;
-
-	pthread_mutex_lock(&j->lock);
-	do {
-		ret = ublksrv_json_write_dev_info(cdev,
-				j->jbuf, j->jbuf_size);
-	} while (ret < 0 && tgt_realloc_jbuf(j));
-	pthread_mutex_unlock(&j->lock);
-
-	return ret;
-}
-
-int ublk_json_write_params(const struct ublksrv_ctrl_dev *cdev,
-		const struct ublk_params *p)
-{
-	struct ublksrv_tgt_jbuf *j = ublksrv_tgt_get_jbuf(cdev);
-	int ret = 0;
-
-	if (!j)
-		return -EINVAL;
-
-	pthread_mutex_lock(&j->lock);
-	do {
-		ret = ublksrv_json_write_params(p, j->jbuf, j->jbuf_size);
-	} while (ret < 0 && tgt_realloc_jbuf(j));
-	pthread_mutex_unlock(&j->lock);
-
-	return ret;
-}
-
-int ublk_json_write_target_base(const struct ublksrv_ctrl_dev *cdev,
-		const struct ublksrv_tgt_base_json *tgt)
-{
-	struct ublksrv_tgt_jbuf *j = ublksrv_tgt_get_jbuf(cdev);
-	int ret = 0;
+#define UBLKSRV_INTERNAL_H_
+#include "ublksrv_priv.h"
```

Comment:
The above need to be removed, and target code won't build if it is done, and `struct ublksrv_ctrl_dev` is actually one private structure, and just its handle is available for user.

Reply:
Thanks for the review.
That needs to be fixed.
The series did hightlight some of the difficulties that need to be un-wrapped in order to do these changes I want.

I will re-work the series over the coming week.